src/Mod/Path/PathScripts/PathFeatureExtensionsGui.py

```diff
@@ -299,16 +301,33 @@ def _initializeExtensions(self, obj):
             self.form.extensionEdit.setDisabled(True)
         self.setExtensions(self.extensions)
 
-    def updateQuantitySpinBoxes(self, index=None):
-        prevValue = self.form.defaultLength.text()
-        self.defaultLength.updateSpinBox()
-        postValue = self.form.defaultLength.text()
+    def _isDefaultLengthExpression(self):
+        """_isDefaultLengthExpression()... Return True if Default Length is determined by an expression."""
+        for prop, __ in self.obj.ExpressionEngine:
+            if prop == "ExtensionLengthDefault":
+                return True
+        return False
 
-        if postValue != prevValue:
-            PathLog.debug("updateQuantitySpinBoxes() post != prev value")
+    def _applyDefaultLengthChange(self, index=None):
+        """_applyDefaultLengthChange(index=None)...
+        Helper method to update Default Length spinbox, and update extensions due to change in Default Length."""
+        self.defaultLength.updateSpinBox()
+        if self.form.defaultLength.text() != self.lastDefaultLength:
+            self.lastDefaultLength = self.form.defaultLength.text()
             self._resetCachedExtensions()  # Reset extension cache because extension dimensions likely changed
             self._enableExtensions()  # Recalculate extensions
 
+    def _defaultLengthChanged(self):
+        """_defaultLengthChanged()... Slot method for determining if a change in Default Length
+        value is determined from an expression edit, or a simple spinbox change. If the former,
+        emit a `editingFinished` signal manually because the Formula Editor window returned
+        a value to the base SpinBox."""
+        if (
+            self._isDefaultLengthExpression()
+            and self.form.defaultLength.text() != self.lastDefaultLength
+        ):
+            self.form.defaultLength.editingFinished.emit()
```

Comment:
Per Qt documentation, the `editingFinished` signal is handled differently than the `valueChanged`.  The `editingFinished` is that same signal that is emitted in a regular QLineEdit when the user has finished editing text and wants the focus to return to the parent widget or task panel.  That is what we need in this case for the extension update code to properly update.  Without the `editingFinished` the code does not update the extensions because focus is still on the QSB - which is why we have had to click outside the QSB after using the Forumula Editor window successfully.

As posted in above code review response, I have incorporated this `valueChanged` to `editingFinished` signal translation into the QuantitySpinBox class where it belongs.  I tested the `StartDepth` inputs (another QSB usage) a number of times and they are behaving correctly and accordingly with these changes.

Thanks for your review and comments and suggestions!